sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -9,54 +10,126 @@ import com.klaviyo.coresdk.networking.requests.KlaviyoRequest
 import com.klaviyo.coresdk.networking.requests.TrackRequest
 
 /**
- * Our main API for the Klaviyo SDK.
- * Where we pass in analytics requests to be sent to the Klaviyo backend
+ * Public API for the core Klaviyo SDK.
+ * Receives configuration, customer data, and analytics requests
+ * to be processed and sent to the Klaviyo backend
  */
 object Klaviyo {
 
     /**
-     * Assigns a user email address to the user info
-     * This email address is stored in memory to define user details during a session and will be
-     * used to autocomplete analytics requests that are missing the identifier in their property maps
+     * Configure Klaviyo SDK with your account's public API Key and application context.
+     * Optionally specify additional behavior customization
      *
-     * Whenever the active user in an app changes (for example via a fresh login) this should be
-     * called to set the new user information in memory.
+     * This must be called to initialize the SDK before using any other functionality
      *
-     * @param email Email address for an active user
+     * @param apiKey - Your Klaviyo account's public API Key
+     * @param applicationContext
+     * @param networkTimeout
+     * @param networkFlushInterval
+     * @param networkFlushDepth
+     * @param networkFlushCheckInterval
+     * @param networkUseAnalyticsBatchQueue
+     * @return
      */
-    fun setUserEmail(email: String) {
+    fun configure(
+        apiKey: String,
+        applicationContext: Context,
+        networkTimeout: Int = KlaviyoConfig.NETWORK_TIMEOUT_DEFAULT,
+        networkFlushInterval: Int = KlaviyoConfig.NETWORK_FLUSH_INTERVAL_DEFAULT,
+        networkFlushDepth: Int = KlaviyoConfig.NETWORK_FLUSH_DEPTH_DEFAULT,
+        networkFlushCheckInterval: Int = KlaviyoConfig.NETWORK_FLUSH_CHECK_INTERVAL,
+        networkUseAnalyticsBatchQueue: Boolean = KlaviyoConfig.NETWORK_USE_ANALYTICS_BATCH_QUEUE,
+    ) = apply {
+        KlaviyoConfig.Builder()
+            .apiKey(apiKey)
+            .applicationContext(applicationContext)
+            .networkTimeout(networkTimeout)
+            .networkFlushInterval(networkFlushInterval)
+            .networkFlushDepth(networkFlushDepth)
+            .networkFlushCheckInterval(networkFlushCheckInterval)
+            .networkUseAnalyticsBatchQueue(networkUseAnalyticsBatchQueue)
+            .build()
+    }
+
+    //region Fluent setters
```

Comment:
Whats this comment for?

Reply:
Oh sorry, I see now

Reply:
Region/endregion makes a collapsible section, its just a code-organization/IDE thing